Objects/dictobject.c

```diff
@@ -6686,21 +6718,15 @@ PyObject_GenericGetDict(PyObject *obj, void *context)
     if (_PyType_HasFeature(tp, Py_TPFLAGS_MANAGED_DICT)) {
         PyDictOrValues *dorv_ptr = _PyObject_DictOrValuesPointer(obj);
         if (_PyDictOrValues_IsValues(*dorv_ptr)) {
```

Comment:
Regarding the atomics in `_PyDictOrValues_IsValues`. I think we need the `*dorv_ptr` access to be atomic -- not the accesses within `_PyDictOrValues_IsValues`.